tests/tests/resource_limiting.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any
+// person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the
+// Software without restriction, including without
+// limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software
+// is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice
+// shall be included in all copies or substantial portions
+// of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
+// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use jsonrpsee::{
+	http_client::HttpClientBuilder,
+	http_server::HttpServerBuilder,
+	proc_macros::rpc,
+	types::{traits::Client, v2::ParamsSer, Error},
+	ws_client::WsClientBuilder,
+	ws_server::{WsServerBuilder, WsStopHandle},
+	RpcModule,
+};
+use tokio::time::sleep;
+
+use std::net::SocketAddr;
+use std::time::Duration;
+
+fn module_manual() -> Result<RpcModule<()>, Error> {
+	let mut module = RpcModule::new(());
+
+	module.register_async_method("say_hello", |_, _| async move {
+		sleep(Duration::from_millis(50)).await;
+		Ok("hello")
+	})?;
+
+	module
+		.register_async_method("expensive_call", |_, _| async move {
+			sleep(Duration::from_millis(50)).await;
+			Ok("hello expensive call")
+		})?
+		.resource("CPU", 3)?;
+
+	module
+		.register_async_method("memory_hog", |_, _| async move {
+			sleep(Duration::from_millis(50)).await;
+			Ok("hello memory hog")
+		})?
+		.resource("CPU", 0)?
```

Comment:
Add comment explaining that this is the same as omitting the resource, i.e. that `0` is the default?